src/main/java/frc/trigon/robot/motorsimulation/MotorSimulation.java

```diff
@@ -0,0 +1,129 @@
+        pidController = new PIDController(config.pidConfigs.kP, config.pidConfigs.kI, config.pidConfigs.kD);
+        voltageCompensationSaturation = config.voltageCompensationSaturation;
+        this.config = config;
+        enablePIDContinuousInput(config.pidConfigs.enableContinuousInput);
+    }
+
+    public void stop() {
+        setVoltage(0);
+    }
+
+    public void setControl(VoltageOut voltageRequest) {
+        setVoltage(voltageRequest.Output);
+    }
+
+    public void setControl(PositionVoltage positionVoltageRequest) {
+        this.positionVoltageRequest = positionVoltageRequest;
+        double voltage = pidController.calculate(getPositionRevolutions() * config.conversionFactor, positionVoltageRequest.Position);
+        setVoltage(voltage);
+    }
+
+    public void setControl(MotionMagicVoltage motionMagicRequest) {
+        this.motionMagicRequest = motionMagicRequest;
+        double pidOutput = profiledPIDController.calculate(getPositionRevolutions() * config.conversionFactor, motionMagicRequest.Position);
+        double feedforwardOutput = calculateFeedforward(config.feedForwardConfigs, Units.rotationsToRadians(motionMagicRequest.Position / config.conversionFactor), profiledPIDController.getGoal().velocity);
+        double output = pidOutput + feedforwardOutput;
+        setVoltage(output);
+    }
+
+    public void setControl(DutyCycleOut dutyCycleRequest) {
+        double voltage = Conversions.compensatedPowerToVoltage(dutyCycleRequest.Output, voltageCompensationSaturation);
+        setVoltage(voltage);
+    }
+
+    public double getVoltage() {
+        return voltage;
+    }
+
+    public double getPosition() {
+        return getPositionRevolutions() * config.conversionFactor;
+    }
+
+    public double getVelocity() {
+        return getVelocityRevolutionsPerSecond() * config.conversionFactor;
+    }
+
+    private void setVoltage(double voltage) {
+        double compensatedVoltage = MathUtil.clamp(voltage, -voltageCompensationSaturation, voltageCompensationSaturation);
+        this.voltage = compensatedVoltage;
+        setInputVoltage(compensatedVoltage);
+    }
+
+    private void enablePIDContinuousInput(boolean enableContinuousInput) {
+        if (enableContinuousInput) {
+            pidController.enableContinuousInput(-(0.5 * config.conversionFactor), 0.5 * config.conversionFactor);
+            profiledPIDController.enableContinuousInput(-(0.5 * config.conversionFactor), 0.5 * config.conversionFactor);
+        } else {
+            pidController.disableContinuousInput();
+            profiledPIDController.disableContinuousInput();
+        }
+    }
+
+    private static void updateRegisteredSimulations() {
+        for (MotorSimulation motorSimulation : REGISTERED_SIMULATIONS) {
+            motorSimulation.setControl(motorSimulation.positionVoltageRequest);
+            motorSimulation.setControl(motorSimulation.motionMagicRequest);
+            motorSimulation.updateMotor();
```

Comment:
1. Update before setting voltage 
2. You can't send two requests, you only want to send one (if one was issued). Make it so only one control can be sent, and that using any other control will make it so you won't send the pid request anymore (i.e. I sent a PositionVoltage request, waited 5 seconds, and then sent a VoltageOut request. The code should try and reach the target position for the five seconds, and then just apply the VoltageOut)